hoodie-client/src/main/java/com/uber/hoodie/func/CopyOnWriteLazyInsertIterable.java

```diff
@@ -61,20 +59,30 @@ public CopyOnWriteLazyInsertIterable(Iterator<HoodieRecord<T>> sortedRecordItr,
     this.hoodieTable = hoodieTable;
   }
 
+  // Used for caching HoodieRecord along with insertValue. We need this to offload computation work to buffering thread.
+  static class HoodieAvroRecordGenResult<T extends HoodieRecord> {
```

Comment:
rename to `HoodieInsertValueGenResult` ? meta question, same thing not needed for MergeHandle, since the buffering does not happen for update path?

Reply:
Done, this happens for merge handle as well, renamed the commit.